packages/components/namespace-store/namespace-store-form.tsx

```diff
@@ -81,42 +81,61 @@ const NamespaceStoreForm: React.FC<NamespaceStoreFormProps> = withHandlePromise<
   NamespaceStoreFormProps & HandlePromiseProps
 >((props) => {
   const { t } = useTranslation();
-  const [nsName, setNsName] = React.useState('');
   const [provider, setProvider] = React.useState(BC_PROVIDERS.AWS);
-  const [pvc, setPVC] = React.useState(null);
-  const [folderName, setFolderName] = React.useState('');
-  const [providerDataState, providerDataDispatch] = React.useReducer(
-    providerDataReducer,
-    initialState
+  const [formDataState, formDataDispatch] = React.useReducer(
+    nssFormReducer,
+    initialStateFormCreation
   );
-  const handleNsNameTextInputChange = (strVal: string) => setNsName(strVal);
+  const handleNsNameTextInputChange = (strVal: string) =>
+    formDataDispatch({ type: 'setName', value: strVal });
   const {
     onCancel,
     className,
     inProgress,
-    errorMessage,
     handlePromise,
     redirectHandler,
     namespace,
+    errorMessage,
   } = props;
 
+  const isFormValid = () => {
+    const {
+      name,
+      accessKey,
+      secretKey,
+      secretName,
+      region,
+      endpoint,
+      target,
+      pvc,
+      pvcFolderName,
+    } = formDataState;
+    return (
+      name &&
+      name.length <= 42 &&
+      (((region || endpoint) &&
+        (secretName || (secretKey && accessKey)) &&
+        target) ||
+        (pvc && pvcFolderName))
+    );
+  };
+
   const onSubmit = (event) => {
     event.preventDefault();
     /** Create a secret if secret ==='' */
-    let { secretName } = providerDataState;
+    let { secretName } = formDataState;
     const promises = [];
-
-    if (!secretName) {
-      secretName = nsName.concat('-secret');
-      const { secretKey, accessKey } = providerDataState;
+    if (provider !== BC_PROVIDERS.FILESYSTEM && !secretName) {
+      secretName = formDataState.name.concat('-secret');
+      const { secretKey, accessKey } = formDataState;
       const secretPayload = secretPayloadCreator(
         provider,
         namespace,
         secretName,
         accessKey,
         secretKey
       );
-      providerDataDispatch({ type: 'setSecretName', value: secretName });
+      formDataState.secretName = secretName;
```

Comment:
```suggestion
      formDataDispatch({ type: 'setSecretName', value: secretName }
```

Reply:
Updating state var directly might be replaced, when formDataDispatch is called somewhere

Reply:
@GowthamShanmugam I tried this, but it is not getting updated.

Reply:
When we update through dispatch it will go and exec your actions. But changes will be reflected in the state var on the next render.

So state var update and use on same function won't be possible. Lets keep this line as formDataDispatch and use  secretName directly (don't use formDataState.secretName after this line).